src/app/external-cluster-wizard/steps/external-cluster/provider/aks/component.ts

```diff
@@ -253,19 +261,36 @@ export class AKSClusterSettingsComponent
         this.vmSizeLabel = this.vmSizes?.length ? VMSizeState.Ready : VMSizeState.Empty;
         this.control(Controls.VmSize).enable();
       });
-      this._getAKSAvailableNodePoolVersionsForCreateMachineDeployment().subscribe(
-        (nodePoolVersions: AKSNodePoolVersionForMachineDeployments[]) => {
-          this.nodePoolVersionsForMD = nodePoolVersions.map(nodePoolVersion => nodePoolVersion.version);
-        }
-      );
+
+      this.nodePoolVersionsForMDLabel = NodePoolVersionState.Loading;
+      this._getAKSAvailableNodePoolVersionsForCreateMachineDeployment()
+        .pipe(tap(_ => this._clearNodePoolVersions()))
+        .subscribe((nodePoolVersions: AKSNodePoolVersionForMachineDeployments[]) => {
+          this.nodePoolVersionsForMD = nodePoolVersions;
+          this.nodePoolVersionsForMDLabel = this.nodePoolVersionsForMD?.length
+            ? NodePoolVersionState.Ready
+            : NodePoolVersionState.Empty;
+          nodePoolVersions.forEach((nodePoolVersion: AKSNodePoolVersionForMachineDeployments) => {
+            if (nodePoolVersion.default) {
+              this.control(Controls.KubernetesVersion).setValue(nodePoolVersion.version);
+            }
+          });
+        });
     } else {
       this.control(Controls.Location)
         .valueChanges.pipe(debounceTime(this._debounceTime))
         .pipe(tap(_ => this._clearVmSize()))
         .pipe(
-          switchMap((location: string) => {
+          switchMap((location: AKSLocation) => {
+            let obs$;
+            const locationValue = this.controlValue(Controls.Location)?.[ComboboxControls.Select];
+            if (locationValue) {
+              obs$ = this._getAKSVmSizes(location?.[ComboboxControls.Select]);
```

Comment:
well it seems like it's working fine but since `location` is of type `AKSLocation` then why you use `[ComboboxControls.Select]` i think here `location` is  `object` `{select: 'string'}`
right ?

Reply:
also update the payload you send for all the field you change it to `km-combobox` to have this value `[ComboboxControls.Select]` 

![Screenshot from 2022-09-14 19-07-37](https://user-images.githubusercontent.com/85109141/190206178-e850b065-3709-4faf-af3d-0c1c427bbcf5.png)
also check the summary page for them 
![Screenshot from 2022-09-14 19-01-49](https://user-images.githubusercontent.com/85109141/190206050-eb30a7c5-67b3-4214-8453-f886cca43729.png)

Reply:
Good catch! and yes we don't need this redundant `?.[ComboboxControls.Select]` we can directly use locationValue